cpp/include/cudf/strings/string_view.cuh

```diff
@@ -43,9 +44,32 @@ __device__ inline size_type characters_in_string(const char* str, size_type byte
   return thrust::count_if(
     thrust::seq, ptr, ptr + bytes, [](uint8_t chr) { return is_begin_utf8_char(chr); });
 }
+
+/**
+ * @brief string value for sentinel which is used in min, max reduction
+ * operators
+ * This sentinel string value is the highest possible valid UTF-8 encoded
+ * character. This serves as identity value for maximum operator on string
+ * values. Also, this char pointer serves as valid device pointer of identity
+ * value for minimum operator on string values.
+ */
+static __constant__ char max_string_sentinel[5]{"\xF7\xBF\xBF\xBF"};
```

Comment:
I have filed a bug in https://github.com/rapidsai/cudf/issues/7455 that is related to this constant which is defined in a header. I would suggest to declare it as `extern` in the .cuh header and define it in a .cu file.

Reply:
We will have to think of some other fix to avoid that bug because as far as I know cudf does not enable `CUDA_SEPARABLE_COMPILATION` and that [prohibits](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-specifiers) adding the extern keyword to __shared__ __device__ or __constant__ variable.

I had added the static keyword to this definition precisely because I wanted to avoid bug #7455 and it looks like it had worked so far. I am open to other ideas to solve this.

Reply:
FYI https://github.com/rapidsai/cudf/pull/7480/files#diff-3b99730080ce6377b3ffa7517e2605a970ccc688355574258b5854fe5639a52bR95